### PR TITLE
For #19955: Add telemetry for home screen recent tabs.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Mugurell** [comment]
@mcarare Since the recents tabs section now contains a normal tab and possibly a media tab this patch needs to be rebased and differentiate between the two types of items shown for which different metrics are to be recorded.

**Comment by mcarare** [comment]
Request for data collection review
---
1. What questions will you answer with this data?
    This data will provide statistics regarding the usage of the new recent tabs on the home screen.
 
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
    This will give us an understanding of how and if the user uses the new section. 

3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
    No alternative is available to answer this question.
 
4. Can current instrumentation answer these questions?
    No.
 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement   Description | Data Collection   Category | Tracking Bug #
-- | -- | --
Show all button clicks| category 2 - Interaction data | https://github.com/mozilla-mobile/fenix/issues/19955
Open Recent tab clicks| category 2 - Interaction data | https://github.com/mozilla-mobile/fenix/issues/19955
Open In progress Media tab button clicks| category 2 - Interaction data | https://github.com/mozilla-mobile/fenix/issues/20393

6. How long will this data be collected? Choose one of the following:
    Data will be collected for 1 year initially (ending on 2022-06-23), with the option to renew at that point. 
 
7. What populations will you measure?
    No filters.
 
8. If this data collection is default on, what is the opt-out mechanism for users?
     Default Glean SDK opt-out mechanism.
 
9. Please provide a general description of how you will analyze this data.
    Glean / Amplitude
 
10. Where do you intend to share the results of your analysis?
      Only on Glean, Amplitude and with mobile teams.
 
11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
      No third-party tool will use this data.

**Comment by travis79** [comment]
> ## Request for data collection review
> 
>     1. What questions will you answer with this data?
>        This data will provide statistics regarding the usage of the new recent tabs on the home screen.
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
>        This will give us an understanding of how and if the user uses the new section.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
>        No alternative is available to answer this question.
> 
>     4. Can current instrumentation answer these questions?
>        No.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Show all button clicks 	category 2 - Interaction data 	#19955
> Open Recent tab clicks 	category 2 - Interaction data 	#19955
> Open In progress Media tab button clicks 	category 2 - Interaction data 	#20393
> 
>     1. How long will this data be collected? Choose one of the following:
>        Data will be collected for 1 year initially (ending on 2022-06-23), with the option to renew at that point.
> 
>     2. What populations will you measure?
>        No filters.
> 
>     3. If this data collection is default on, what is the opt-out mechanism for users?
>        Default Glean SDK opt-out mechanism.
> 
>     4. Please provide a general description of how you will analyze this data.
>        Glean / Amplitude
> 
>     5. Where do you intend to share the results of your analysis?
>        Only on Glean, Amplitude and with mobile teams.
> 
>     6. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
>        No third-party tool will use this data.

# Data Review 

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the Glean Dictionary.

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the app menu.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection scheduled to end in 1 year (2022-06-23).

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction Data

5) Is the data collection request for default-on or default-off?

Default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

**Review by travis79 — APPROVED** [review]
data-review+ only, doesn't include code-review.

File: app/src/main/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabController.kt [hunk 1]
```diff
@@ -35,15 +40,27 @@ interface RecentTabController {
...
     private val selectTabUseCase: SelectTabUseCase,
-    private val navController: NavController
+    private val navController: NavController,
+    private val metrics: MetricController,
+    private val store: BrowserStore
 ) : RecentTabController {
 
     override fun handleRecentTabClicked(tabId: String) {
+
+        when (tabId) {
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabController.kt:50]: nit: We could save some tabs iterations if we only check for `inProgressMediaTab` and record `OpenRecentTab` in all other cases.

File: app/src/test/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabControllerTest.kt [hunk 2]
```diff
@@ -68,6 +73,7 @@ class RecentTabControllerTest {
             selectTabUseCase.selectTab.invoke(tab.id)
             navController.navigate(R.id.browserFragment)
         }
+        verify { metrics.track(Event.OpenRecentTab) }
```
Comment (Mugurell) [inline app/src/test/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabControllerTest.kt:76]: nit: This check can sit in the above `verify` block.

File: app/src/test/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabControllerTest.kt [hunk 3]
```diff
@@ -80,5 +86,6 @@ class RecentTabControllerTest {
                 null
             )
         }
+        verify { metrics.track(Event.ShowAllRecentTabs) }
```
Comment (Mugurell) [inline app/src/test/java/org/mozilla/fenix/home/recenttabs/controller/RecentTabControllerTest.kt:89]: nit: This check can sit in the above `verify` block.

**Review by Mugurell — APPROVED** [review]

**Review by Mugurell — APPROVED** [review]

**Comment by mcarare** [comment]
Unrelated unit test fail ShortcutsSuggestionProviderTest. callbacks are triggered when suggestions are clicked.

**Pull request merged**